packages/@aws-cdk/aws-iam/lib/oidc-provider/index.ts

```diff
@@ -44,20 +47,16 @@ async function onUpdate(event: AWSLambda.CloudFormationCustomResourceUpdateEvent
 
   const providerArn = event.PhysicalResourceId;
 
-  // if thumbprints changed, we can update in-place, but bear in mind that if the new thumbprint list
-  // is empty, we will grab it from the server like we do in CREATE
-  const oldThumbprints = (event.OldResourceProperties.ThumbprintList || []).sort();
-  if (JSON.stringify(oldThumbprints) !== JSON.stringify(thumbprints)) {
-    const thumbprintList = thumbprints.length > 0 ? thumbprints : [await external.downloadThumbprint(issuerUrl)];
-    external.log('updating thumbprint list from', oldThumbprints, 'to', thumbprints);
-    await external.updateOpenIDConnectProviderThumbprint({
-      OpenIDConnectProviderArn: providerArn,
-      ThumbprintList: thumbprintList,
-    });
-
-    // don't return, we might have more updates...
```

Comment:
the `downloadThumbprint` needs to always happen in case no thumbprints are specified, so it must be moved out of the `if` block. I figured i'd just simplify this altogether and call  `updateOpenIDConnectProviderThumbprint` always, even if nothing changed - potentially running some no-ops once in a while.